assets/js/editor/transform/ClassicBlockTransformer.js

```diff
@@ -59,19 +59,47 @@ class ClassicBlockTransformer {
 		}
 	}
 
+	/**
+	 * Determines the state of a given promise.
+	 *
+	 * @param {Promise} promise - The promise whose state is to be determined.
+	 * @returns {Promise<string>} - A promise that resolves to a string indicating the state:
+	 * - "pending" if the original promise is still pending,
+	 * - "fulfilled" if the original promise has been fulfilled,
+	 * - "rejected" if the original promise has been rejected.
+	 */
+	async getPromiseState(promise) {
+		try {
+			await Promise.race([promise, Promise.reject(new Error('pending'))]);
+			return 'fulfilled';
+		} catch (e) {
+			if (e.message === 'pending') {
+				return 'pending';
+			}
+			throw e;
+		}
+	}
+
 	/**
 	 * If the specified block is a freeform / classic block, replaces it
 	 * with corresponding Gutenberg blocks
 	 *
 	 * @param {object} block The current block object
 	 */
-	transform(block) {
+	async transform(block) {
 		if (this.isFreeformBlock(block)) {
-			this.wp.data
+			const didReplacePromise = this.wp.data
 				.dispatch('core/block-editor')
 				.replaceBlocks(block.clientId, this.blockHandler(block));
 
-			this.didTransform = true;
+			try {
+				const dispatchStatus = await this.getPromiseState(didReplacePromise);
+				if (dispatchStatus === 'fulfilled') {
+					this.didTransform = true;
+				}
+			} catch (e) {
+				// Do nothing
```

Comment:
This feels like it will silently hide errors during block transformation. Do we need the try-catch here?